pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -150,7 +155,24 @@ func (r *ReconcileChaosEngine) Reconcile(request reconcile.Request) (reconcile.R
 		}
 		return reconcile.Result{}, err
 	}
+	deletetimeStamp := engine.Instance.ObjectMeta.GetDeletionTimestamp()
 
+	if engine.Instance.Spec.EngineStatus == "stop" ||
```

Comment:
can you move this block in a separate funcition

Reply:
It's just an if condition , that is calling some other function. I don't feel the need for passing it to separate function.

Reply:
But it is violating the BCH link due to the number of line of code